scripts/components/NewPropertyForm.jsx

```diff
@@ -0,0 +1,155 @@
+      tenantEmail: '', // User entered tenant email
+      dataUrl: '', // base64 encoding of the user selected image.
+    }
+  },
+
+  /**
+   * Event handler for capturing in the input field state on each keypress.
+   * @param  {String} field The identifier for the input field.
+   */
+  onChange(field, event) {
+    this.setState({ [field]: event.target.value });
+  },
+
+  /**
+   * Form submission event handler. Sends a request to the server to add the
+   * repair request, and updates the repair requests if successful.
+   */
+  onSubmit(event) {
+    event.preventDefault();
+    event.stopPropagation();
+
+    // API call to add new property
+    Api.addNewProperty({
+      data: {
+        streetAddress: this.state.streetAddress,
+        suburb: this.state.suburb,
+        postCode: this.state.postCode,
+        ownerEmail: this.state.ownerEmail,
+        tenantEmail: this.state.tenantEmail,
+        dataUrl: this.state.dataUrl,
+      },
+      callback: (err, response) => {
+        if (err) {
+          return;
+        }
+
+        // Clear the form
+        this.setState({
+          streetAddress: '',
+          suburb: '',
+          postCode: '',
+          ownerEmail: '',
+          tenantEmail: '',
+          dataUrl: '',
+        });
+      }
+    });
+  },
+
+  /**
+   * Callback for a user selecting a file for upload. Reads the file as a base64
+   * encoded data URI and stores this in component state.
+   */
+  onFileSelected(event) {
+    var reader = new FileReader(); // File API
+    var file = event.target.files[0];
+
+    reader.onload = upload => this.setState({ dataUri: upload.target.result });
+    reader.readAsDataURL(file);
+  },
+
+  render() {
+    var { streetAddress, suburb, postCode, ownerEmail, tenantEmail, dataUrl } = this.state;
+    var errorMessage;
+    return (
+      <div>
+        <form style={style.form} onSubmit={this.onSubmit}>
+          <h2>Add New Property</h2>
+          <div style={style.error}> { errorMessage } </div>
+          <TextField
+            value={streetAddress}
+            multiLine={true}
+            onChange={this.onChange.bind(this, 'streetAddress')}
+            floatingLabelText="Street Address" />
+          <TextField
+            value={suburb}
+            multiLine={true}
+            onChange={this.onChange.bind(this, 'suburb')}
+            floatingLabelText="Suburb" />
+          <TextField
+            value={postCode}
+            multiLine={true}
+            onChange={this.onChange.bind(this, 'postCode')}
+            floatingLabelText="Post Code" />
+          <TextField
+            value={ownerEmail}
+            multiLine={true}
+            onChange={this.onChange.bind(this, 'ownerEmail')}
+            floatingLabelText="Owner Email" />
+          <TextField
+            value={tenantEmail}
+            multiLine={true}
+            onChange={this.onChange.bind(this, 'tenantEmail')}
+            floatingLabelText="Tenant Email" />
+          <TextField
```

Comment:
This needs to be a `<input type='file' onChange={this.onFileSelected}/>` if you want a data URL encoding of a selected image.